### DISCUSSION TITLE
URI the whole folder, is it possible?

### DISCUSSION THREAD

**Discussion by Parakovsky** [savonet/liquidsoap#3700] [User Support]
Since I've read the book already passed few months, as mentioned before anywhere the results far exceeds any expectations, except in one field... can't figure out for the best of me how to queue the whole folder, say Madonna made a new album and operator want to queue it to the broadcast right away, the URI identificator seems to only be able to parse individual files/playlists, so the solution seems to be to create another queue with an empty playlist inside say tmp/playlist.m3u directory and then point to it trough the right order so it takes precedence over other lists when the file paths are loaded there, but it seems not to work, or only works half of the time. Here's the syntaxis for playlists used:

s = replaygain(playlist(reload_mode="watch", "/storage/Audio")) - this is the main one
z = replaygain(playlist(reload_mode="watch", "/tmp/playlist.m3u")) - this is the custom thing me thinking about to load playlists in, the order at this point doesn't really matters
live = input.harbor("tuziknet", port=1420, password="COMPROMISED176") - here's my live dj setup, password is not real and so is the port
queue = request.queue() - classic request queue me use when put individual tracks/sets from telnet (bind into shortcut in Sway)
backup = replaygain(playlist("/storage/Audio/SOUNDS OF ALIENS IN SPACE BUT VERY MUTED"))

s = fallback (track_sensitive=false, [live, queue, z, s, backup]) - the whole ensemble.

So what the problem is that the playlist Z is not get activated when the data inside it updated. Ideally will be ace to just get rid of it altogether and then just use some weird URI think that will be able to parse it straight to queue thing, like /storage/Audio/Madonna - Crazy Dazy Love Journey/ and it will get all the files from it. This seems to be not possible at the moment... 

Please advice, and thank you for reading. Will also post on the list but it seems to be rather lonely there. Salute, Aleksas

PS this is my initial draft guess can be used for TLDR purposes

After months of working inside 'Soap, still not clear how, for examplewill give you a usage scenario: have a custom randomized playlist inside/storage/Audio have a new folder there called /storage/Audio/Mum_Dancewith tracks 1, 2, 3, 4... etc. is there a chance it could be programmedthat one can feed *entire* folder into already running server instancewithout changing anything in the config and it will start playing thetracks from it? so far my understanding of URI is that it only acceptone file except when you have a playlist or somethnig, which itselfis one file as well.

**Comment by toots** [comment]
Hi!

Have you tried using `file.ls` to list all the files in the folder and queue them one by one?

**Reply by Parakovsky** [reply]
Thank you Toots for enlightening me, I check this property out and report back! Have a great day ahead!
